low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -22,28 +22,220 @@
+                    break
+        if not fru_found:
+            raise ResourceMapError(
+                errno.EINVAL,
+                "Health provider doesn't have support for'{rpath}'.")
+        return info
+
+    @staticmethod
+    def get_node_details(node):
+        """
+        Parse node information and returns left string and instance.
+
+        Example
+            "storage"    -> ("storage", "*")
+            "storage[0]" -> ("storage", "0")
+        """
+        res = re.search(r"(\w+)\[([\d]+)\]|(\w+)", node)
+        inst = res.groups()[1] if res.groups()[1] else "*"
+        node = res.groups()[0] if res.groups()[1] else res.groups()[2]
+        return node, inst
+
+    def get_data_template(self, uid, is_fru: bool):
+        """Get the health data template."""
+        return {
+            "uid": uid,
+            "fru": str(is_fru).lower(),
+            "last_updated": int(time.time()),
+            "health": {
+                "status": "",
+                "description": "",
+                "recommendation": "",
+                "specifics": []
+            }
+        }
+
+    def set_health_data(self, obj: dict, status, description=None,
+                        recommendation=None):
+        """Set the given or default health data as per health status."""
+        good_state = (status == "OK")
+        if not description:
+            description = "%s is %sin good state" % (
+                            obj.get("uid"),
+                            '' if good_state else 'not ')
+        if not recommendation:
+            recommendation = 'None' if good_state\
+                             else "Contact Seagate Support."
+
+        obj["health"].update({
+            "status": status,
+            "description": description,
+            "recommendation": recommendation
+        })
+
+    def get_sas_hba_info(self):
+        """Return the s SAS-HBA information."""
+        return self.get_sas_ports_info("sas_hba")
+
+    def get_sas_ports_info(self, leaf_node="sas_ports"):
+        """Return the s SAS ports information."""
+        sas_hba_data = []
+        health_data = []
+        for sas_rid in self.SAS_RESOURCE_ID:
+            r_data = self.get_data_template(sas_rid, False)
+            sas_ports_dict = self.sysfs.get_phy_negotiated_link_rate()
+            sas_ports_dict = sorted(sas_ports_dict.items(),
+                                    key=lambda kv: int(kv[0].split(':')[1]))
+            health = []
+            if sas_ports_dict:
+                for rid, rate in sas_ports_dict:
+                    r_data["health"]["specifics"].append({
+                        "resource_id": f"{sas_rid}:{rid.strip()}",
+                        "negotiated_link_rate": rate.strip()
+                    })
+                    if "Gbit".lower() in rate.strip().lower():
+                        health.append("OK")
+
+            # By some logic set health status, description, recommendation
```

Comment:
Dont retain such comments, 'By some logic ..'. Make it a useful one

Reply:
removing, forgot to remove it was a placeholder comment.